packages/@o3r/localization/schematics/add-localization-key/index.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  chain,
+  noop,
+  Rule,
+  SchematicContext,
+  Tree
+} from '@angular-devkit/schematics';
+import {
+  applyEsLintFix,
+  getO3rComponentInfo,
+  isO3rClassComponent
+} from '@o3r/schematics';
+import { dirname, resolve } from 'node:path';
+import * as ts from 'typescript';
+import type { NgAddLocalizationKeySchematicsSchema } from './schema';
+
+const getLocalizationInformation = (componentPath: string, tree: Tree) => {
+  const componentSourceFile = ts.createSourceFile(
+    componentPath,
+    tree.readText(componentPath),
+    ts.ScriptTarget.ES2015,
+    true
+  );
+
+  const o3rClass = componentSourceFile.statements.find((statement): statement is ts.ClassDeclaration => ts.isClassDeclaration(statement) && isO3rClassComponent(statement))!;
+  let localizationJsonPath: string | undefined;
+  let defaultTranslationVariableName: string | undefined;
+  let translationsVariableName: string | undefined;
+  let translationsVariableType: string | undefined;
+  o3rClass.members.forEach((member) => {
```

Comment:
What happens if o3rClass is undefined ? 
Should we throw saying that the component is not an otter component? 
Question aside: Is it mandatory to be an otter component ?

Reply:
`getO3rComponentInfo` is called before this method and will ensure we have an otter component

> Is it mandatory to be an otter component ?
That what we have done everywhere and if it is an angular component we propose to convert the component into an otter one before when https://github.com/AmadeusITGroup/otter/issues/480 will be done